nexosim-util/src/joiners.rs

```diff
@@ -0,0 +1,135 @@
+//! Thread handlers.
+//!
+//! This module contains threading related utilities.
+//!
+
+use std::fmt;
+use std::marker::PhantomData;
+use std::thread::{JoinHandle, Result as ThreadResult};
+
+use nexosim::simulation::Scheduler;
+
+type DropAction<T> = Box<dyn FnOnce(ThreadResult<T>) + Send + 'static>;
+
+/// A thread guard.
+///
+/// This is a thread guard that automatically joins the thread in the
+/// destructor. Additionally, a custom drop action can be specified, which will
+/// be executed on the thread's execution result during the guard's
+/// destruction. The thread can also be explicitly joined using the `join`
+/// method.
+pub struct ThreadJoiner<T> {
+    /// A phantom data marker.
+    _data: PhantomData<T>,
```

Comment:
Is the marker required here? My guess is that `T` is used in the other fields, and so the marker is redundant.

Reply:
Yes, thanks. This is a leftover of the previous design.